### DISCUSSION TITLE
Is there a way to force HtmlRenderer to close tags?

### DISCUSSION THREAD

**Discussion by divnull** [vsch/flexmark-java#552] [Q&A] [answered]
Is there an option or something that will force HtmlRenderer to close any tags left open by bad input? For example, if the user submits markdown that contains an opening ``` block, but never adds the close, like:

`````
Some text
```
Preformatted text
Doesn't close the preformat block
`````

When this renders the output will contain a tag to open the preformatted block, but not one to close it. So, if I embed that result into some page, the open tag will play havoc with that page. As the parser is in the best position to know that it opened that tag, it would be nice to tell it "you know, close all those open tags at the end for safety, even if the input sucks".

**Comment by vsch** [comment, marked as answer]
@divnull, it seems like a bug in `HtmlRenderer`. I will be addressing accumulated PRs and issues in the next couple of weeks.
